File: Observer/TypeScript/src/displays/ForecastDisplay.ts
import { Observer } from '../interfaces/Observer';
import { DisplayElement } from '../interfaces/DisplayElement';
import WeatherData from '../data/WeatherData';

class ForecastDisplay implements Observer, DisplayElement {
  private weatherData: WeatherData;
  private currentPressure = 29.92;
  private lastPressure: number;

  constructor(weatherData: WeatherData) {
    this.weatherData = weatherData;
    weatherData.registerObserver(this);
  }
  display() {
    console.log('Forecast: ');
    if (this.currentPressure > this.lastPressure) {
      console.log('Improving weather on the way!');
    } else if (this.currentPressure === this.lastPressure) {
      console.log('More of the same');
    } else if (this.currentPressure < this.lastPressure) {
      console.log('Watch out for cooler, rainy weather');
    }
  }
  update() {
    this.lastPressure = this.currentPressure;
    this.currentPressure = this.weatherData.getPressure();
    this.display();
  }
}

export default ForecastDisplay;

File: Observer/Java/src/displays/ForecastDisplay.java
package eu.svetlin.design.patterns.observer.displays;

import eu.svetlin.design.patterns.observer.data.WeatherData;
import eu.svetlin.design.patterns.observer.interfaces.DisplayElement;
import eu.svetlin.design.patterns.observer.interfaces.Observer;


public class ForecastDisplay implements Observer, DisplayElement {
    private WeatherData weatherData;
    private float currentPressure = 29.92f;
    private float lastPressure;

    public ForecastDisplay(WeatherData weatherData){
        this.weatherData = weatherData;
        weatherData.registerObserver(this);
    }

    @Override
    public void display(){
        System.out.print("Forecast: ");
        if (currentPressure > lastPressure) {
            System.out.println("Improving weather on the way!");
        } else if (currentPressure == lastPressure) {
            System.out.println("More of the same");
        } else if (currentPressure < lastPressure) {
            System.out.println("Watch out for cooler, rainy weather");
        }

    }

    @Override
    public void update(){
        lastPressure = currentPressure;
        currentPressure = weatherData.getPressure();
        display();
    }

}

